### ISSUE TITLE
FR: Make "swipe down to close menu/window" optional

### ISSUE THREAD

**Issue by mergen3107** [koreader/koreader#10138]
**Does your feature request involve difficulty completing a task? Please describe.**
Can please make "swipe down to close menu/window" optional?
At least, that I could turn it off completely.

Optionally, make it just another dispatcher item that can be reassigned to other gestures (if needed).

**Additional context**
The context is on Android. I have my screenshot button in the Android notification area that is summoned by swiping down from the top edge.

Coincidentally, that is the same swipe direction to close a window in KOReader. I was in History, so it closed each time I tried to open notification area. This is why I intentionally made my top/bottom menus to open upon a tap only.

As an alternative, I resorted to use KOReader's built-in (and also not configurable) screenshot shortcut - tap two opposite corners. However, with Onyx Boox Kon-Tiki 2's rounded corners that was also a challenge :D 

If I get annoyed by that too much, I will resort to SSH-based command for taking screenshot, although that won't be ideal when I would like to capture a screenshot on the move...

**Comment by hius07** [comment]
One more screenshot gesture: long diagonal swipe.

**Comment by mergen3107** [comment]
Thanks! This somehow never worked for me on Onyx. I definitely could do it on Kindle and Kobo Clara HD, which both have the screen bezels, but not Onyx’es flush screen.

**Comment by jonnyl2** [comment]
@mergen3107 You can also tap diagonally opposite corners simultaneously to take screenshots, e.g. top left and bottom right.

**Comment by mergen3107** [comment]
@jonnyl2 
Yes, but as I said in the OP, this doesn’t work reliably on Onyx with rounded corners. The glass has black painted bezel underneath, which seems also blocking the touch function right in the corners.

**Comment by mergen3107** [comment]
Any user patch I could make to disable swipe down to close menus?

**Comment by hius07** [comment]
https://github.com/koreader/koreader/blob/39d54956ec42e97e0c3e57ea3ff994b6ebe57e31/frontend/ui/widget/menu.lua#L1339-L1344

**Comment by mergen3107** [comment]
@hius07 
I haven't figured out how to replace functions in patches. Can I `gsub` `direction = "south"` to something like `direction = "southsouth"` (some nonsense that doesn't get recognized)?

**Comment by hius07** [comment]
https://github.com/koreader/koreader/issues/9277#issuecomment-1172932205

**Comment by jonnyl2** [comment]
@mergen3107:

No guarantees but this seems to work as a user patch:

`2-menu_noswipedown.lua`

```lua
local Menu = require("ui/widget/menu")
local BD = require("ui/bidi")
local UIManager = require("ui/uimanager")

function Menu:onSwipe(arg, ges_ev)
    local direction = BD.flipDirectionIfMirroredUILayout(ges_ev.direction)
    if direction == "west" then
        self:onNextPage()
    elseif direction == "east" then
        self:onPrevPage()
    elseif direction == "south" then
        if not self.no_title then
        do end
    elseif direction == "north" then
        do end
    else 
        UIManager:setDirty(nil, "full")
    end
end
end

**Comment by mergen3107** [comment]
@jonnyl2 
Thank you! I quickl tested this on Onyx Boox, seems to work.

However, I switched to a Kindle recently, so won't be using this.

I'll add the `user-patch available` tag for those interested.
